client/app/dbaas/logs/detail/streams/logs-streams.service.js

```diff
@@ -106,6 +205,104 @@ class LogsStreamsService {
                 return quota;
             }).catch(this.ServiceHelper.errorHandler("logs_streams_quota_get_error"));
     }
+
+    getCompressionAlgorithms () {
+        return this.compressionAlgorithms;
+    }
+
+    getStorageDurations () {
+        return this.storageDurations;
+    }
+
+    getSubscribedOptions (serviceName) {
+        return this.LogsOptionsService.getStreamSubscribedOptions(serviceName);
+    }
+
+    /**
+     * returns number of streams assigned for given operation
+     *
+     * @param {any} operationId
+     * @returns number of streams assigned for given operation
+     * @memberof LogsStreamsService
+     */
+    _getNumberOfStreamsAssigned (serviceName, operationId) {
+        const defer = this.$q.defer();
+        if (this.optionStreamMap) {
+            defer.resolve(this.optionStreamMap[operationId] ? this.optionStreamMap[operationId] : 0);
+        } else {
+            this.getStreams(serviceName).then(() => defer.resolve(this.optionStreamMap[operationId] ? this.optionStreamMap[operationId] : 0));
+        }
+        return defer.promise;
+    }
+
+    /**
+     * add additional data to stream before sending back to controller
+     * 1. asynchronously gets notifications of a stream
+     * 2. asynchronously gets archives of a stream
+     * 3. updates operationStreamMap to get number of streams assigned to each operation
+     *
+     * @param {any} serviceName
+     * @param {any} stream
+     * @returns stream object after adding notifications
+     * @memberof LogsStreamsService
+     */
+    _transformStream (serviceName, stream) {
+        stream.notifications = [];
+        // asynchronously fetch all notification of a stream
+        this.getNotifications(serviceName, stream.streamId)
+            .then(notifications => {
+                stream.notifications = notifications;
+            });
+        // asynchronously fetch all archives of a stream
+        this.getArchives(serviceName, stream.streamId)
+            .then(archives => {
+                stream.archives = archives;
+            });
+        if (stream.optionId && this.optionStreamMap[stream.optionId]) {
+            this.optionStreamMap[stream.optionId]++;
+        } else if (stream.optionId) {
+            this.optionStreamMap[stream.optionId] = 1;
+        }
+        return stream;
+    }
+
+    /**
+     * handles success state for create, delete and update streams.
+     * Repetedly polls for operation untill it returns SUCCESS message.
+     *
+     * @param {any} serviceName
+     * @param {any} operation, operation to poll
+     * @param {any} successMessage, message to show on UI
+     * @returns promise which will be resolved to operation object
+     * @memberof LogsStreamsService
+     */
+    _handleSuccess (serviceName, operation, successMessage) {
+        this.poller = this._pollOperation(serviceName, operation);
+        return this.poller.$promise
```

Comment:
I'm not sure the service should be responsible to init polling.. we normally do that in the controller... but I understand that's you can't really do that in this case so I don't know what to tell you.. polling sucks >.<

Reply:
yes it does. I dont have any place to put it. Worst thing is many of this will come in future in other pages ;(.

Reply:
Let that be for now, we'll see what we can do later I guess..